### DISCUSSION TITLE
transmission port is closed

### DISCUSSION THREAD

**Discussion by ghost** [transmission/transmission#2448] [Q&A]
Hello everyone, I have a problem recently I have done some uploads and they believe they are not sending data and not downloading this saying that the door is closed, being that in windows 7 everything worked normal, does anyone know why this happens in linux?

I got to do this command on iptables
sudo iptables -I INPUT -p tcp --dport PORT_NUMBER -j ACCEPT

according to what i read on ask ubuntu

[Ask Ubuntu](https://askubuntu.com/questions/405487/transmission-says-port-is-closed-but-seeding-is-happening)

**Comment by neheb** [comment]
I think the problem is different. If I go to the website manually, it works fine. Transmission is probably having issues getting to the site.

**Comment by tiagoboldt** [comment]
Same here. Using UPnP and random port at start. Restarted transmission and checked http://www.canyouseeme.org/. Port is always reported as open. Using transmission web UI always says port is closed. Latest version, arch linux.

**Comment by zme-ul** [comment]
I'm on Windows - Transmission's port test returns **CLOSED** even tho when checking from an outside source (http://www.canyouseeme.org/ for example) it returns **OPEN**

tested on multiple systems with uPnP enabled and disabled and with the port manually forwarded in the router

even disabled Windows' Firewall - no joy

**Comment by MoBattah** [comment]
Question for all of you, is your connection IPv4 or IPv6?

**Comment by tiagoboldt** [comment]
IPv6

**Comment by zme-ul** [comment]
I have IPv6 enabled

**Comment by MoBattah** [comment]
That's the problem, I believe. I can verify later but I believe Transmission is checking IPv4.
Check out issue #381 

Try forcing an IPv4 connection and see if it changes anything.

**Comment by neheb** [comment]
I did some debugging on this issue. Transmission seems to want CURL_CA_BUNDLE to be set to w/e it is (/etc/ssl/certs/ca-certificates.crt in my case). After that, it manages to connect to the test web page.

And get 0 bytes as a response... No idea why.

**Comment by neheb** [comment]
Found it. If libcurl is built with --disable-verbose, port checking fails. If libcurl is built with --enable-verbose, port checking works fine.

@mikedld any insight?

edit: Scratch that. it's --disable-zlib that's the problem. I see the issue in web.c

**Comment by TimeZlicer** [comment]
As per suggestion by MoBattah to force IPv4

Append the following line to **/etc/hosts**
`87.98.162.88 portcheck.transmissionbt.com`

**Reply by Risiamu** [reply]
This is still the best solution in 2022

**Comment by avibrazil** [comment]
Adding that entry to /etc/hosts solves the problem. I hope now Transmission broadcasts me as a seeder.

**Comment by mobal** [comment]
Adding that entry to ```/etc/hosts``` solved my problem too. This issue is still open.

After I replaced my old router with the new one I have **IPv4** and **IPv6** address. Since then, this problem is present.

**Comment by neheb** [comment]
To people having issues with IPv6, have you tried running transmission with debugging to see the issue? Such as:

transmission-daemon --log-debug

**Comment by RomLecat** [comment]
Same issue here, when IPv6 is enabled the port is reported "Closed" even if it's open (both on IPv4 and IPv6). Not a huge deal, but annoying for testing.

**Comment by davidporos92** [comment]
Forcing to IPv4 by adding the entry to the host also solved the problem for me. Great thanks! 🎆 👏

**Comment by ploink** [comment]
Same problem here.
I suggest someone in charge of the DNS for transmissionbt.com to fix it:


```
$ host portcheck.transmissionbt.com
portcheck.transmissionbt.com is an alias for vm4.transmissionbt.com.
vm4.transmissionbt.com has address 87.98.162.88
vm4.transmissionbt.com has IPv6 address 2001:41d0:c:5ac:5::1

$ dig -t aaaa portcheck.transmissionbt.com

; <<>> DiG 9.11.4-P2-RedHat-9.11.4-13.P2.fc28 <<>> -t aaaa portcheck.transmissionbt.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19147
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;portcheck.transmissionbt.com.  IN      AAAA

;; ANSWER SECTION:
portcheck.transmissionbt.com. 300 IN    CNAME   vm4.transmissionbt.com.
vm4.transmissionbt.com. 300     IN      AAAA    2001:41d0:c:5ac:5::1

;; Query time: 26 msec
;; SERVER: 192.168.178.1#53(192.168.178.1)
;; WHEN: Sat Feb 09 13:18:10 CET 2019
;; MSG SIZE  rcvd: 103
```

Replacing the CNAME record for portcheck with an A record to the direct IP address 87.98.162.88 ensures clients don't connect using ipv6.

**Comment by snixel** [comment]
The DNS issue still persists today.
I also found out that on ipv6 only the TCP port of Transmission is open, UDP is closed on ipv6, while on ipv4 both are open.
Can anyone else test and confirm this?

**Comment by ploink** [comment]
All 4 ports are open on my system:
$ lsof -i -P -n | grep 63979
transmiss 31503 joost   15u  IPv4 58385369      0t0  TCP *:63979 (LISTEN)
transmiss 31503 joost   16u  IPv6 58385371      0t0  TCP *:63979 (LISTEN)
transmiss 31503 joost   17u  IPv4 58385372      0t0  UDP *:63979
transmiss 31503 joost   18u  IPv6 58385374      0t0  UDP [2001:...............]:63979
$ rpm -q transmission-daemon
transmission-daemon-2.94-2.fc28.x86_64

But, transmission only creates IPv4 port forwards, no ipv6 pinholes.
If you want ipv6 port forwards, create them manually.
https://forum.transmissionbt.com/viewtopic.php?t=19275

The port check simply does not work with ipv6. :

$ curl -i **-6** http://portcheck.transmissionbt.com/63979
HTTP/1.1 400 Bad Request
Server: nginx/1.10.3 (Ubuntu)
Date: Sat, 06 Jul 2019 17:50:37 GMT
Content-Type: text/html; charset=ISO-8859-1
Transfer-Encoding: chunked
Connection: keep-alive

$ curl -i **-4** http://portcheck.transmissionbt.com/63979
HTTP/1.1 200 OK
Server: nginx/1.10.3 (Ubuntu)
Date: Sat, 06 Jul 2019 17:55:28 GMT
Content-Type: text/html; charset=ISO-8859-1
Transfer-Encoding: chunked
Connection: keep-alive

1

(returned 1, meaning port is checked open)

**Comment by snixel** [comment]
That's odd, on my system only TCP shows up on ipv6

```
# netstat -lnp | grep -w '51413'
tcp        0      0 0.0.0.0:51413           0.0.0.0:*               LISTEN      661/transmission-da
tcp        0      0 :::51413                    :::*                       LISTEN      661/transmission-da
udp       0      0 0.0.0.0:51413           0.0.0.0:*                                661/transmission-da
```

**Comment by dausruddin** [comment]
Wow. I took an overnight figuring out what I did wrong to my server setup. Took so many times comparing configurations with my old server.
Then a few moment ago I installed qbittorrent and my port works. Turn out transmission is the problem, not my server lol.

**Comment by ghost** [comment]
unfortunately, the issue is still present
```
$ transmission-gtk -v
transmission-gtk 2.94 (d8e60ee44f)
```

I love this torrent client but for the first time I'm thinking about using another one, I only got my port showed as "opened" after forcing use of IPv4.
Thank you @TimeZlicer

**Comment by nicolasdev** [comment]
Same issue for me

```
lsof -i -P -n | grep 4700
transmiss  1337 debian-transmission   16u  IPv6  19298      0t0  TCP *:4700 (LISTEN)
transmiss  1337 debian-transmission   17u  IPv4  19299      0t0  UDP *:4700
cnid_meta  1353                root    3u  IPv4  19954      0t0  TCP 127.0.0.1:4700 (LISTEN)

```

What did you do to force IPv4 @fhenriquedev ?

**Comment by kz6wk9** [comment]
> Same issue for me
> 
> ```
> lsof -i -P -n | grep 4700
> transmiss  1337 debian-transmission   16u  IPv6  19298      0t0  TCP *:4700 (LISTEN)
> transmiss  1337 debian-transmission   17u  IPv4  19299      0t0  UDP *:4700
> cnid_meta  1353                root    3u  IPv4  19954      0t0  TCP 127.0.0.1:4700 (LISTEN)
> ```
> 
> What did you do to force IPv4 @fhenriquedev ?

@nicolasdev
Append this to your **/etc/hosts**
`87.98.162.88 portcheck.transmissionbt.com`

**Comment by aleczk** [comment]
I'm kinda retarded, where can I find the /etc/hosts on Windows? Checked the %appdata% (Local) and the Program Files but no deal.

**Comment by dausruddin** [comment]
@aleczk quick search lead me to `C:\Windows\System32\drivers\etc\hosts`

**Comment by nackstein** [comment]
I had the same problem.
My ISP give my public IPv4 and IPv6.
transmission use UPnP and I see the rule on the gateway but still got "port closed"
adding  87.98.162.88	portcheck.transmissionbt.com to the /etc/hosts (and restartin transmission-daemon) solved my problem

**Comment by BelowAvgJoe** [comment]
This is still an issue. 
I've added 87.98.162.88 portcheck.transmissionbt.com to /etc/hosts
added by port to ufw
confirmed forwarding in router
confirmed open in canyouseeme.org
port is still showing closed in transmission listening port.

**Comment by kisglaci** [comment]
Having the same issue still on Ubuntu 20.04

**Comment by txVLN** [comment]
I'm running transmission in a jail on FreeBSD 11.3 (FreeNAS), and have confirmed that Transmission is listening and traffic is arriving inside the jail. Have confirmed with https://www.ipfingerprints.com/portscan.php that the port is open. Appending /etc/hosts did not work for me. Comments in other threads mention this is a 3.0 issue and reverting to 2.94 solves the problem.

**Comment by Pentaphon** [comment]
> Comments in other threads mention this is a 3.0 issue and reverting to 2.94 solves the problem.

I can confirm. This really needs to be fixed in the next version. 3.0 has many issues, especially on Windows.

**Comment by nackstein** [comment]
running transmission-daemon 3.00 on raspberry pi os right now. 
with previous 2.94 I had to add name resolution for portcheck.transmissionbt.com in /etc/hosts (see my previous comment).
after upgrade to 3.00 (I upgraded whole os, now it's based on debian bulleye) I don't have any problem related to port open check.
raspberry have local ip 192.168.x.x and ISP router have public IPv4 (I disabled IPv6 due to some issues). I have UPnP enabled both on transmission-daemon and ISP router. and I have flagged randomize port on start option.
Only once I saw "port closed" and a simple restart of transmission-daemon solved it. 
never tested on windows.

**Comment by ertyu** [comment]
As mentioned by others above, the check simply fails with active IPv6, the portcheck server doesn't respond properly over IPv6. The dns hosts mapping forces an IPv4 check. Still an issue, but should be simple enough to fix the portcheck server.

**Comment by bugith** [comment]
Still same issue in Ubuntu 22.04

**Comment by ploink** [comment]
I guess this issue was not specific enough and became too old. It has never been picked up, just converted to discussion. So I created a new issue #3102.

**Comment by Marquis-of-Sandwiches** [comment]
2025 and having this problem with transmission-gtk 4.0.6 on Debian Trixie. (Testing as of time of writing) 
Like all the other posts, checking the port from the outside shows it as open. (You need to port map/NAT/open a port on your firewall.)
Add the /etc/hosts referenced in the thread.
And then force the computer to preferring ipv4. 
The file you change is here:  /etc/gai.conf.  (gai == get address info)  
This is in the file to uncomment.  Here are the important lines.
label  ::1/128       0
label  ::/0          1
label  2002::/16     2
label ::/96          3
label ::ffff:0:0/96  4
precedence  ::1/128       50
precedence  ::/0          40
precedence  2002::/16     30
precedence ::/96          20
precedence ::ffff:0:0/96  100
Taken from here: https://weblog.lkiesow.de/20220311-make-linux-prefer-ipv4.html

Something about the mix of ipv4/ipv6 port checking is not working right.  I assume I know why this works, but, honestly, I don't know for sure.
